infrastructure/src/test/java/org/corfudb/infrastructure/LogReplicationServerTest.java

```diff
@@ -126,19 +150,77 @@ public void testHandleLeadershipQuery() {
 
         doReturn(SAMPLE_HOSTNAME).when(context).getLocalEndpoint();
 
-        // verify the response when not the leader
-        lrServer.getHandlerMethods().handle(request, null,
-            mockServerRouter);
+        //verify the response when leader and sessionManager knows about the session
+        lrServer.getHandlerMethods().handle(request, null, mockServerRouter);
         ArgumentCaptor<ResponseMsg> argument = ArgumentCaptor.forClass(ResponseMsg.class);
-        verify(mockServerRouter).sendResponse(argument.capture());
+        verify(mockServerRouter, times(1)).sendResponse(argument.capture());
+        Assertions.assertThat(argument.getValue().getPayload().getLrLeadershipResponse().getIsLeader()).isTrue();
 
-        lrServer.getHandlerMethods().handle(request, null,
-            mockServerRouter);
+        // verify the response when leader but sessionManager does not know about the session yet
+        replicationContext = new LogReplicationContext(mock(LogReplicationConfigManager.class),
```

Comment:
how does this cause the session manager to be unaware of the session?

Reply:
The LogReplicationServer accepts a reference to "sessions" object in the sessionManager. So I am emulating the sessionManager not knowing about the session by passing an empty set in "sessions" place.